pkg/dockerregistry/server/client/client.go

```diff
@@ -0,0 +1,162 @@
+package client
+
+import (
+	restclient "k8s.io/client-go/rest"
+	kcoreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+
+	authclientv1 "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/authorization/v1"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+
+	"github.com/openshift/origin/pkg/client"
+
+	imageclientv1 "github.com/openshift/origin/pkg/image/generated/clientset/typed/image/v1"
+	userclientv1 "github.com/openshift/origin/pkg/user/generated/clientset/typed/user/v1"
+)
+
+// RegistryClient provides Origin and Kubernetes clients to Docker Registry.
+type RegistryClient interface {
+	// Client returns the authenticated client to use with the server.
+	Client() (Interface, error)
+	// ClientFromToken returns a client based on the user bearer token.
+	ClientFromToken(token string) (Interface, error)
+}
+
+// Interface contains client methods that registry use to communicate with
+// Origin or Kubernetes API.
+type Interface interface {
```

Comment:
Interesting.  I may not have done it, but we'll see how it plays out.

Reply:
@deads2k we needed a wrapper if we want to potentially add metrics for the API calls, I think that was originally @legionus idea